services/galley/test/integration/API.hs

```diff
@@ -3665,25 +3673,113 @@ putReceiptModeWithRemotesOk = do
         @?= EdConvReceiptModeUpdate
           (ConversationReceiptModeUpdate (ReceiptMode 43))
 
+postTypingIndicatorsV2 :: TestM ()
+postTypingIndicatorsV2 = do
+  c <- view tsCannon
+  g <- view tsUnversionedGalley
+
+  alice <- randomUser
+  bob <- randomUser
+
+  aliceL <- qualifyLocal alice
+  bobL <- qualifyLocal bob
+
+  connectUsers alice (singleton bob)
+
+  conv <- decodeConvId <$> postO2OConv alice bob Nothing
+  lcnv <- qualifyLocal conv
+
+  WS.bracketR2 c alice bob $ \(wsAlice, wsBob) -> do
+    post
+      ( g
+          . paths ["v2", "conversations", toByteString' conv, "typing"]
+          . zUser alice
+          . zConn "conn"
+          . zType "access"
+          . json (TypingData StartedTyping)
+      )
+      !!! const 200 === statusCode
+
+    void . liftIO $
+      WS.assertMatchN (5 # Second) [wsAlice, wsBob] $ \n ->
+        wsAssertTyping (qUntagged lcnv) (qUntagged aliceL) StartedTyping n
+
+    post
+      ( g
+          . paths ["v2", "conversations", toByteString' conv, "typing"]
+          . zUser bob
+          . zConn "conn"
+          . zType "access"
+          . json (TypingData StoppedTyping)
+      )
+      !!! const 200 === statusCode
+
+    void . liftIO $
+      WS.assertMatchN (5 # Second) [wsAlice, wsBob] $ \n ->
+        wsAssertTyping (qUntagged lcnv) (qUntagged bobL) StoppedTyping n
+
 postTypingIndicators :: TestM ()
 postTypingIndicators = do
+  domain <- viewFederationDomain
+  c <- view tsCannon
   g <- viewGalley
+
   alice <- randomUser
   bob <- randomUser
+
+  aliceL <- qualifyLocal alice
+  bobL <- qualifyLocal bob
+
   connectUsers alice (singleton bob)
+
   conv <- decodeConvId <$> postO2OConv alice bob Nothing
+  lcnv <- qualifyLocal conv
+
+  WS.bracketR2 c alice bob $ \(wsAlice, wsBob) -> do
+    -- alice to bob
```

Comment:
Shouldn't the comment on this line be swapped with the one on line 3754?

Reply:
Good catch!